src/Input/DownshiftSelect/Select.tsx

```diff
@@ -0,0 +1,355 @@
+import React, { HTMLProps, useState } from 'react';
+import { useCombobox, UseComboboxProps } from 'downshift';
+
+import * as internalComponents from './SelectStyle';
+import { ArrowDownIcon, ArrowUpIcon, CloseCircleSolidIcon } from '../..';
+import { isFunction, isUndefined, noop } from 'lodash';
+import { useId } from 'react-id-generator';
+
+export interface Item {
+  value: string;
+  label: string;
+  disabled?: boolean;
+  [x: string]: any;
+}
+
+export const defaultTransformFunction = (items: Item[], inputValue: string) =>
+  items.filter(item =>
+    item.label.toLowerCase().includes(inputValue.toLowerCase())
+  );
+
+export const itemToString = (item: Item) => (item ? item.label : '');
+
+export const defaultEmptyListText =
+  'No results found. Try another keyword to search for.';
+
+// These are our custom props for our subcomponents. We need this because
+// sometimes we want to make additional props available to the subcomponent,
+// e.g. a Item subcomponent needs to know the item it represents.
+export type ItemProps = {
+  item: Item;
+};
+
+// These are the actual types for the subcomponents. Each subcomponent takes
+// its own, custom props (defined by us) AND the props that correspond to the
+// intrinsic html element which the subcomponent "extends".
+type ContainerType = HTMLProps<HTMLDivElement>;
```

Comment:
Oh, will it constrain the `Container` to be a `div` only?

Reply:
I think nothing will break if it's a different component. Though I don't see which other component could be used here.